mapadroid/worker/WorkerMITM.py

```diff
@@ -198,72 +212,56 @@ def __update_injection_settings(self):
-                                break
-                        if data_requested != LatestReceivedType.UNDEFINED:
-                            break
-                    if data_requested is None or data_requested == LatestReceivedType.UNDEFINED:
-                        self.logger.debug("No spawnpoints in data requested")
-                        time.sleep(1)
-                elif mode in ["raids_mitm"]:
-                    self.logger.debug("Checking raids_mitm data")
-                    for data_extract in latest_data['payload']['cells']:
-                        for forts in data_extract['forts']:
-                            # if there's location in latest, the distance has
-                            # already been checked in MITMBase
-                            if forts['id'] and (latest.get("location", None) or
-                                                self._check_data_distance(latest_data['payload']['cells'])):
-                                self.logger.debug("Got proper fort data for raids")
-                                data_requested = latest_data
-                                break
-                        if data_requested != LatestReceivedType.UNDEFINED:
-                            break
-                    if data_requested is None or data_requested == LatestReceivedType.UNDEFINED:
-                        self.logger.debug("No forts in data received: {}", latest_data)
-                        time.sleep(0.5)
-                    else:
-                        self.logger.debug("Got data requested: {}", data_requested)
-                else:
-                    self.logger.warning("No mode specified to wait for - this should not even happen...")
-                    time.sleep(0.5)
+        # proto has previously been received, let's check the timestamp...
+        mode = self._mapping_manager.routemanager_get_mode(self._routemanager_name)
+        timestamp_of_proto: float = latest_proto_entry.get("timestamp", None)
+        self.logger.debug("Latest timestamp: {} vs. timestamp waited for: {} of proto {}",
+                          datetime.fromtimestamp(timestamp_of_proto), datetime.fromtimestamp(timestamp),
+                          proto_to_wait_for)
+        if timestamp_of_proto < timestamp:
+            self.logger.debug("latest timestamp of proto {} ({}) is older than {}", proto_to_wait_for,
+                              timestamp_of_proto, timestamp)
+            # TODO: timeout error instead of data_error_counter? Differentiate timeout vs missing data (the
+            # TODO: latter indicates too high speeds for example
+            return type_of_data_found, data_found
+
+        latest_proto_data: dict = latest_proto_entry.get("values", None)
+        if latest_proto_data is None:
+            return LatestReceivedType.UNDEFINED, data_found
+        latest_proto = latest_proto_data.get("payload")
+        if mode in ["mon_mitm", "iv_mitm"]:
```

Comment:
The if/elif has a lot of duplicate code. We should convert this to a pure function with parameters to reduce duplicate code.